clang/lib/CodeGen/CGExprConstant.cpp

```diff
@@ -1886,6 +1896,29 @@ llvm::Constant *ConstantEmitter::emitForMemory(CodeGenModule &CGM,
     return Res;
   }
 
+  if (destType->isBitIntType()) {
+    if (CGM.getTypes().typeRequiresSplitIntoByteArray(destType, C->getType())) {
+      // Long _BitInt has array of bytes as in-memory type.
+      // So, split constant into individual bytes.
+      ConstantAggregateBuilder Builder(CGM);
+      llvm::Type *DesiredTy = CGM.getTypes().ConvertTypeForMem(destType);
+      llvm::Type *LoadStoreTy =
```

Comment:
It would seem instead of invoking `convertTypeForLoadStore` we could use the bit width of the in-memory type and sign-/zero- extend the `APInt Value`, something like.
```
llvm::APInt Value;
if destType is unsigned 
  Value = CI->getValue().zext(Width);
else
 Value = CI->getValue().sext(Width);
```

Reply:
Using `convertTypeForLoadStore` is aligned with other similar places and IMO seems more generalized approach.

Reply:
Right.  Momchil, your description is actually a totally reasonable way of understanding what `convertTypeForLoadStore` *means* for `_BitInt`: the load/store type is an integer type that has the same width as the in-memory type.  We're just recognizing that as a more general concept.